java/code/src/com/redhat/rhn/domain/contentmgmt/PackageFilter.java

```diff
@@ -87,10 +87,10 @@ private static <T> T getField(Package pack, String field, Class<T> type) {
 
     private static boolean checkNameAndArch(String field, String value, Package pack) {
         if (field.equals("nevr")) {
-            return value.replaceAll("(.*)-(.*:)?(.*)-(.*)", "$1").equals(pack.getPackageName().getName());
+            return value.replaceAll("^(.*)-([^:]*+:)?([^-]*+)-([^-]*+)$", "$1").equals(pack.getPackageName().getName());
         }
         else if (field.equals("nevra")) {
-            return value.replaceAll("(.*)-(.*:)?(.*)-(.*)\\.(.*)", "$1$5")
+            return value.replaceAll("^(.*)-([^:]*+:)?([^-]*+)-([^-]*)\\.([^.-]*+)$", "$1$5")
```

Comment:
Shouldn't the third group be `[^.]*` rather? Probably the last group can be `.*`: wouldn't backtrack